services/repository/files/temp_repo.go

```diff
@@ -77,8 +77,8 @@ func (t *TemporaryUploadRepository) Clone(branch string) error {
 }
 
 // Init the repository
-func (t *TemporaryUploadRepository) Init(objectFormat git.ObjectFormat) error {
-	if err := git.InitRepository(t.ctx, t.basePath, false, objectFormat); err != nil {
+func (t *TemporaryUploadRepository) Init(objectFormatName string) error {
```

Comment:
Why replace objectFormat with a string? I guess this can be asked for all instances replaced. To me having a type is stronger guarantee of correctness than a string.

Reply:
Because only the name is necessary to create the repository. Following minial parameter principle.

Reply:
I guess it depends how the minimal parameter principle is defined. I would argue that `objectFormat` is much more restrictive and hence minimal parameter that can be used than `string`. In its current iteration, it's purely static type.

string, on the other hand, can be any value to compiler and these guarantees of correctness disappear.

Reply:
We can add a validate here. And for objectFormat, it can also be `nil`.

Reply:
You are correct about the `nil`.  This is what was problematic when in the SHA256 patch - I couldn't rely that the model.objectFormat was not `nil` in all circumstances. This then results in needing to parse the objectFormat of actual repository which is much heavier operation.

The database needs to store name anyway so I guess I don't have strong objections here.